src/hb-ot-post-table.hh

```diff
@@ -55,6 +55,91 @@ struct postV2Tail
     return_trace (glyphNameIndex.sanitize (c));
   }
 
+  void unpackPStrings (unsigned max_index, hb_map_t& index_map) const
+  {
+    if (max_index < 258) return;
+    const UnsizedArrayOf<const HBUINT8>& names = StructAfter<const UnsizedArrayOf<const HBUINT8>> (glyphNameIndex);
+    unsigned num = max_index - 257;
+    unsigned length = 0;
+
+    for (unsigned i = 0; i < num; i++)
+    {
+      index_map.set (i+258, length);
+      unsigned name_length = names[length];
+      length += 1 + name_length;
+    }
+  }
+
+  template<typename Iterator,
+           hb_requires (hb_is_iterator (Iterator))>
+  bool serialize (hb_serialize_context_t *c,
+                  Iterator it,
+                  const hb_map_t& index_map) const
+  {
+    TRACE_SERIALIZE (this);
+    auto *out = c->start_embed (this);
+    if (unlikely (!c->check_success (out))) return_trace (false);
+    if (!out->glyphNameIndex.serialize (c, + it
+                                           | hb_map (hb_second)))
+        return_trace (false);
+
+    const UnsizedArrayOf<const HBUINT8> &names = StructAfter<const UnsizedArrayOf<const HBUINT8>> (glyphNameIndex);
```

Comment:
No sanitization is done on the strings array currently. So it's not safe to access the strings memory directly. You could instead use the accelerator_t::find_glyph_name() which does bounds checking for you.